### ISSUE TITLE
logger instead of print

### ISSUE THREAD

**Issue by smilesun** [marrlab/DomainLab#131]

**Comment by Car-la-F** [comment]
@smilesun 
I have some questions about this issue. 

I looked into the logging tutorial of python (https://docs.python.org/3/howto/logging.html#basic-logging-tutorial). They mention two variants, replacing the prints with logging, but still writing everything to command line or logging into a file. I guess we like to use the logging to file, but I still want to be sure.

Shall we have the logging only for the benchmark or also for single runs from the command line? 

For me it's the first time working with logging, hence I have some more basic questions: 
- What do we like the user to specify? Shall the logging level be hard coded or shall it be a command line argument? 
- Do you have an idea at which level the logger can be placed? 
- Do I have to replace all prints by a logging.info('...') or is there a faster way (I mean I don't mind doing the long way, but I'm open for a better solution).

**Comment by smilesun** [comment]
Q3:
1. if you use Linux you could try
“Ack”or sed with regular expression
2. If you use py charm, there is a function called refactor if I
remember correctly
3. It is definitely not a good idea to do them file by file

We could discuss about the solution before we start doing this issue.

**Comment by Car-la-F** [comment]
I have used a refactor function in pycharm before, but it did only consider the file which is currently open, hence this would still be file by file. But maybe this is not the function you meant in 2.

Shall we have a meeting about this? 
Currently I do work in a shared office during weekdays, hence I would prefer to have a meeting in the morning, before I go there (maybe around 8-9:30) or after I come back (Tuesday or Friday after 17 o'clock). If this does not fit for you I will also find a different solution.

**Comment by smilesun** [comment]
find ./ -type f -exec sed -i 's/get_device(args.nocu)/get_device(args)/' {} \;

will check all files under current directory  ./
and change all strings "get_device(args.nocu)" to "get_device(args)"

**Comment by Car-la-F** [comment]
@smilesun,

I worked on this issue again. I found out that snakemake has the option to pass a logfile. But this feature is already implemented for our benchmark:

https://github.com/marrlab/DomainLab/blob/23cdb6c8aee1bbfb199a3578758c8ca557326162/run_benchmark_local_conf_seed2_gpus.sh#LL63C171-L63C178

The logfile is saved in `"zoutput/logs"` with the fiename beeing a time stamp. It simply writes all prints to a file, so there is no filtering happening right now (but there is also no need to change the prints to logger.debug, logger.info, ...)
